src/cdi_CPEloss/Tabular_CP_Eloss.hh

```diff
@@ -0,0 +1,100 @@
+//----------------------------------*-C++-*----------------------------------//
+/*!                                                                              
+ * \file   cdi_CPEloss/Tabular_CP_Eloss.hh                                     
+ * \author Ben R. Ryan                                                        
+ * \date   2019 Nov 4                                              
+ * \brief  Tabular_CP_Eloss class definition.                                   
+ * \note   Copyright (C) 2016-2019 Triad National Security, LLC.                 
+ *         All rights reserved. */
+//---------------------------------------------------------------------------//
+
+#ifndef __cdi_CPEloss_Tabular_CP_Eloss_hh__
+#define __cdi_CPEloss_Tabular_CP_Eloss_hh__
+
+#include "cdi/CPCommon.hh"
+#include "cdi/CPEloss.hh"
+#include "ds++/Assert.hh"
+#include "ds++/DracoArray.hh"
+#include "ds++/DracoMath.hh"
+#include <cmath>
+#include <fstream>
+#include <string>
+#include <vector>
+
+namespace rtt_cdi_cpeloss {
+//===========================================================================//
+/*!                                                                              
+ * \class Tabular_CP_Eloss                                                      
+ *                                                                               
+ * \brief Derived rtt_cdi::CPEloss class for tabular eloss.                     
+ *        This class implements the interface found in cdi/CPEloss.hh for        
+ *        the case where CP energy loss data is in tabular form, stored 
+ *        in a file.                
+ *                                                                               
+ */
+//===========================================================================//
+
+class Tabular_CP_Eloss : public rtt_cdi::CPEloss {
+public:
+  typedef std::vector<double> sf_double;
+
+private:
+  std::string filename;
+  std::ifstream file;
+
+  uint32_t n_energy;          // Number of gridpoints in projectile energy
```

Comment:
Change comment trailing style from `// comment` to `//!< comment`.

Reply:
Can any be const or constexpr?

Reply:
Comment style changed, but as far as I can tell none of these member variables can be `const`/`constexpr` because they are assigned during the constructor from values in the data file.